gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/identifierfirst/IdentifierFirstLoginEndpoint.java

```diff
@@ -0,0 +1,119 @@
+    private static final String ALLOW_REGISTER_CONTEXT_KEY = "allowRegister";
+    private static final String ALLOW_PASSWORDLESS_CONTEXT_KEY = "allowPasswordless";
+    private static final String REGISTER_ACTION_KEY = "registerAction";
+    private static final String WEBAUTHN_ACTION_KEY = "passwordlessAction";
+
+    private final Domain domain;
+    private final BotDetectionManager botDetectionManager;
+
+    public IdentifierFirstLoginEndpoint(TemplateEngine templateEngine, Domain domain, BotDetectionManager botDetectionManager) {
+        super(templateEngine);
+        this.domain = domain;
+        this.botDetectionManager = botDetectionManager;
+    }
+
+    @Override
+    public void handle(RoutingContext routingContext) {
+        final Client client = routingContext.get(ConstantKeys.CLIENT_CONTEXT_KEY);
+        prepareContext(routingContext, client);
+        renderLoginPage(routingContext, client);
+    }
+
+    private void prepareContext(RoutingContext routingContext, Client client) {
+        final HttpServerRequest request = routingContext.request();
+        // remove sensible client data
+        routingContext.put(ConstantKeys.CLIENT_CONTEXT_KEY, new ClientProperties(client));
+        // put domain in context data
+        routingContext.put(ConstantKeys.DOMAIN_CONTEXT_KEY, domain);
+        // put request in context
+        EvaluableRequest evaluableRequest = new EvaluableRequest(new VertxHttpServerRequest(request.getDelegate(), true));
+        routingContext.put(REQUEST_CONTEXT_KEY, evaluableRequest);
+
+        LoginSettings loginSettings = LoginSettings.getInstance(domain, client);
+        var optionalSettings = ofNullable(loginSettings).filter(Objects::nonNull);
+        routingContext.put(ALLOW_REGISTER_CONTEXT_KEY, optionalSettings.map(LoginSettings::isRegisterEnabled).orElse(false));
+        routingContext.put(ALLOW_PASSWORDLESS_CONTEXT_KEY, optionalSettings.map(LoginSettings::isPasswordlessEnabled).orElse(false));
+
+        // put error in context
+        final String error = request.getParam(ConstantKeys.ERROR_PARAM_KEY);
+        final String errorDescription = request.getParam(ConstantKeys.ERROR_DESCRIPTION_PARAM_KEY);
+        routingContext.put(ConstantKeys.ERROR_PARAM_KEY, error);
+        routingContext.put(ConstantKeys.ERROR_DESCRIPTION_PARAM_KEY, errorDescription);
+
+        // put parameters in context (backward compatibility)
+        Map<String, String> params = new HashMap<>(evaluableRequest.getParams().toSingleValueMap());
+        params.put(ConstantKeys.ERROR_PARAM_KEY, error);
+        params.put(ConstantKeys.ERROR_DESCRIPTION_PARAM_KEY, errorDescription);
+        routingContext.put(ConstantKeys.PARAM_CONTEXT_KEY, params);
+
+        final MultiMap queryParams = RequestUtils.getCleanedQueryParams(request);
+        routingContext.put(ACTION_KEY, resolveProxyRequest(request, routingContext.get(CONTEXT_PATH) + "/login", queryParams, true));
```

Comment:
I would prefer to have a dedicated POST /login/identifier route for that screen.

POST /login is for authentication process
where POST /login/identifier will be used only for the redirection purpose

Reply:
`POST /login/identifier` does not exist, today this just a get to the Login endpoint